pkgs/tools/networking/openconnect/default.nix

```diff
@@ -1,12 +1,39 @@
-{ lib, stdenv, fetchurl, pkg-config, openssl ? null, gnutls ? null, gmp, libxml2, stoken, zlib, fetchgit, darwin } :
+{ lib, stdenv, fetchurl, fetchgit
+, pkg-config, makeWrapper
+, openssl ? null, gnutls ? null
+, gmp, libxml2, stoken, zlib
+, nettools, gawk, openresolv, coreutils, gnugrep
+, PCSC
+} :
 
 assert (openssl != null) == (gnutls == null);
 
-let vpnc = fetchgit {
-  url = "git://git.infradead.org/users/dwmw2/vpnc-scripts.git";
-  rev = "c0122e891f7e033f35f047dad963702199d5cb9e";
-  sha256 = "11b1ls012mb704jphqxjmqrfbbhkdjb64j2q4k8wb5jmja8jnd14";
-};
+let
+  os = if stdenv.isDarwin then "Darwin" else "Linux";
+
+  vpnc = stdenv.mkDerivation {
+    name = "vpnc-scripts-c0122e891f7";
+    src = fetchgit {
+      url = "git://git.infradead.org/users/dwmw2/vpnc-scripts.git";
+      rev = "c0122e891f7e033f35f047dad963702199d5cb9e";
+      sha256 = "11b1ls012mb704jphqxjmqrfbbhkdjb64j2q4k8wb5jmja8jnd14";
+    };
+
+    nativeBuildInputs = [ makeWrapper ];
+
+    installPhase = ''
+      mkdir -p $out/bin
+      cp vpnc-script $out/bin
+
+      substituteInPlace $out/bin/vpnc-script \
+        --replace "which" "type -P" \
+        --replace "/sbin/resolvconf" "${openresolv}/bin/resolvconf"
+
+      wrapProgram $out/bin/vpnc-script \
+        --set OS ${os} \
+        --prefix PATH : "${lib.makeBinPath [ nettools gawk openresolv coreutils gnugrep ]}"
+    '';
```

Comment:
Might another package want to use those? Maybe we should move it then to top-level.

Reply:
@SuperSandro2000 I agree with you that we should move that into an own top-level package.

Reply:
sure; done.